src/commands.c

```diff
@@ -307,13 +311,30 @@ const CommandSpec *CommandSpecGet(const RedisModuleString *cmd)
  * the aggregate flags of all commands. If a command is not listed in the
  * command table, use default_flags.
  */
-unsigned int CommandSpecGetAggregateFlags(RaftRedisCommandArray *array, unsigned int default_flags)
+unsigned int CommandSpecGetAggregateFlags(RedisModuleCtx *ctx, RaftRedisCommandArray *array, unsigned int default_flags)
 {
     unsigned int flags = 0;
     for (int i = 0; i < array->len; i++) {
         const CommandSpec *cs = CommandSpecGet(array->commands[i]->argv[0]);
         if (cs) {
             flags |= cs->flags;
+            if (cs->flags & CMD_SPEC_FLAGS) {
+                if (!strncmp(cs->name, "eval", strlen("eval")) && array->commands[i]->argc > 1) {
+                    uint64_t sub_flags;
+                    if (RedisModule_GetScriptBodyFlags(ctx, array->commands[i]->argv[1], &sub_flags) == REDISMODULE_OK) {
+                        if (!(sub_flags && (REDISMODULE_SCRIPT_FLAG_NO_WRITES | REDISMODULE_SCRIPT_FLAG_ALLOW_OOM))) {
+                            flags |= CMD_SPEC_DENYOOM;
+                        }
+                    }
+                } else if (!strncmp(cs->name, "evalsha", strlen("evalsha")) && array->commands[i]->argc > 1) {
+                    uint64_t sub_flags;
+                    if (RedisModule_GetScriptSHAFlags(ctx, array->commands[i]->argv[1], &sub_flags) == REDISMODULE_OK) {
```

Comment:
It looks good to me. Just have two questions:
- Do we need to pass `ctx` to these new functions? Maybe, we won't ever need it. We can discuss with others.
- As you mentioned before, do functions require any special treatment?

Reply:
ok, realized we probably don't need context (was from error message returing, but perhaps not needed anymore, we would just return 0 for flags in case of error and later code would handle the error).

functions would need their own Get*Flags functions which haven't been written, but would work similarly I'd expect.